File: 90 - Dev/vault_root.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""Корень хранилища (vault) для инструментария 90 - Dev.

Расположение может быть двумя (оба поддерживаются, см. комментарий в vault-root.js):

  1) инструментарий внутри хранилища — <vault>/90 - Dev/* → корень = <dev>/..;
  2) инструментарий рядом с хранилищем — <dev>/../Lecture-Graph-Vault/*.

Раньше скрипты жёстко считали корнем вариант 2 и падали в самом репозитории
(ENOENT на .obsidian/plugins/lecture-graph/data.json)."""
import os


def vault_root(dev_dir):
    nested = os.path.abspath(os.path.join(dev_dir, "..", "Lecture-Graph-Vault"))
    if os.path.isdir(os.path.join(nested, ".obsidian")):
        return nested
    return os.path.abspath(os.path.join(dev_dir, ".."))


if __name__ == "__main__":
    print(vault_root(os.path.dirname(os.path.abspath(__file__))))

File: 90 - Dev/vault-root.js
/* Корень хранилища (vault) для инструментария 90 - Dev.

   Возможны два расположения, и оба должны работать:

     1) инструментарий ВНУТРИ хранилища — так устроен этот репозиторий:
        <vault>/90 - Dev/*  →  корень = <dev>/.. (там лежит .obsidian);
     2) инструментарий РЯДОМ с хранилищем — так гоняли пайплайн, когда сам
        репозиторий был только набором скриптов:
        $HOME/90 - Dev/*  +  $HOME/Lecture-Graph-Vault/*  →  корень = <dev>/../Lecture-Graph-Vault.

   Скрипты раньше были жёстко привязаны к варианту 2 и падали (ENOENT) в варианте 1,
   то есть в самом репозитории. Здесь расположение определяется один раз по наличию
   .obsidian, поэтому копия инструментария «едет» вместе с хранилищем и не ломается. */
const fs = require("fs");
const path = require("path");

module.exports = function vaultRoot(devDir) {
  const nested = path.resolve(devDir, "..", "Lecture-Graph-Vault");
  if (fs.existsSync(path.join(nested, ".obsidian"))) return nested;
  return path.resolve(devDir, "..");
};
